data/tutorials/guides/0tt_05_s_expressions.md

```diff
@@ -0,0 +1,267 @@
+
+Here are a few examples:
+
+```lisp
+()
+(a)
+(a b)
+(a b c)
+(a (b 0) (c 1.2))
+```
+
+Each of the individual symbols are atoms. An **atom** represents a basic data value. Atoms can be of different types, including integers, floating-point numbers, strings, or Booleans. In OCaml, atoms are represented as their corresponding OCaml data types. For example:
+
+  + Integer: `42`
+  + Float: `3.14`
+  + String: `"Hello, world!"`
+  + Boolean: `true`
+
+
+Inside each set of parentheses is a nested list of S-Expressions. A nested list is used to represent the **hierarchical structure of S-Expressions**. Each element can be an atom or another nested list. The elements within the list are separated by spaces.
+
+In OCaml, S-Expressions are defined as follows:
+
+```ocaml
+type sexp =
+  | Atom of string
+  | List of sexp list (** Nested list. *)
+```
+
+This type and the basic operations associated to this type are defined in the [sexplib](https://github.com/janestreet/sexplib) package.
+
+Note: the syntax of [Dune configuration files](https://dune.readthedocs.io/en/latest/reference/lexical-conventions.html) is similar to S-expressions.
+
+## Serialising
+
+> Serialisation is the process of translating a data structure or object state into a format that can be stored (e.g., files or data buffers) or transmitted and reconstructed later.
+
+Here it means we can translate OCaml data types into S-expressions and then into a more machine readable format.
+
+### Constructing
+
+The [Core](https://github.com/janestreet/core) package provides many ways to deal with S-expressions, a `Sexp` module to deal with the S-expressions themselves, and `sexp_of_t` / `t_of_sexp` functions provided by data type modules (like `String`, `List`, etc.) that allow users to convert OCaml data to and from S-expressions.
+
+The provided examples are intended to be excuted in UTop. Here is the initial setup:
+
+```ocaml
+# require "sexplib";;
+# open Sexplib;;
+```
+
+To manually build S-expressions we can combine the two constructors `Atom` and `List` presented above. To build the S-expression representing `(1)`: 
+```ocaml
+# Sexp.Atom "1";;
+- : Type.t = Sexplib.Sexp.Atom "1"
+```
+To build the S-expression representing `(1 2)`:
+```ocaml
+# Sexp.List [Atom "1"; Atom "2"];;
+- : Type.t = Sexplib.Sexp.List [Sexplib.Sexp.Atom "1"; Sexplib.Sexp.Atom "2"]
+```
+
+### Printing and Formatting
+
+To display the representation of the S-expressions, formatting functions like `Sexp.pp` are provided:
+
+```ocaml
+# Format.printf "%a\n" Sexp.pp Sexp.(List [List [Atom "1"; Atom "hello"]]);;
+((1 hello))
+- : unit = ()
+```
+
+To convert an S-expression to a string, simply use `Sexp.to_string`:
+```ocaml
+# Sexp.to_string Sexp.(List [List [Atom "1"; Atom "hello"]]);;
```

Comment:
Given that atoms are strings I think it would be instructive to include a string a that has a space.